### PR TITLE
Replace ASButtonState with ASControlState

### PR DESCRIPTION
**Pull request by binl**
following the UIControl design, removed ASButtonState from ASButtonNode and added ASControlState to ASControlNodes, also added a selectedState to all ASControlNodes.

@appleguy @connor

### CODE REVIEW

File: AsyncDisplayKit/ASControlNode.h [hunk 1]
```diff
@@ -55,6 +62,12 @@ typedef NS_OPTIONS(NSUInteger, ASControlNodeEvent)
  */
 @property (nonatomic, readonly, assign, getter=isHighlighted) BOOL highlighted;
 
+/**
+ @abstract Indicates whether or not the receiver is highlighted.
+ @discussion This is set automatically when the receiver is tapped.
+ */
+@property (nonatomic, assign, getter=isSeleted) BOOL selected;
```
Comment (appleguy) [inline AsyncDisplayKit/ASControlNode.h]: @binl missing a c in "isSelected"

File: AsyncDisplayKit/ASControlNode.h [hunk 2]
```diff
@@ -34,6 +34,13 @@ typedef NS_OPTIONS(NSUInteger, ASControlNodeEvent)
   ASControlNodeEventAllEvents         = 0xFFFFFFFF
 };
 
+typedef NS_OPTIONS(NSUInteger, ASControlState) {
```
Comment (appleguy) [inline AsyncDisplayKit/ASControlNode.h]: This looks nice and clean, but - where was it declared before?

File: AsyncDisplayKit/ASControlNode.h [hunk 3]
```diff
@@ -34,6 +34,13 @@ typedef NS_OPTIONS(NSUInteger, ASControlNodeEvent)
...
+    ASControlStateNormal       = 0,
+    ASControlStateHighlighted  = 1 << 0,                  // used when ASControlNode isHighlighted is set
+    ASControlStateDisabled     = 1 << 1,
+    ASControlStateSelected     = 1 << 2,                  // used when ASControlNode isSeleted is set
```
Comment (appleguy) [inline AsyncDisplayKit/ASControlNode.h]: Also typo for isSelected - check for other instances

File: AsyncDisplayKit/ASControlNode.h (hunk 2 above)
Comment (binl) [inline AsyncDisplayKit/ASControlNode.h]: it was in ASButtonNode.h

File: AsyncDisplayKit/ASControlNode.h (hunk 1 above)
Comment (binl) [inline AsyncDisplayKit/ASControlNode.h]: oops..

File: AsyncDisplayKit/ASButtonNode.mm [hunk 4]
```diff
@@ -17,10 +17,12 @@ @interface ASButtonNode ()
   
   NSAttributedString *_normalAttributedTitle;
   NSAttributedString *_highlightedAttributedTitle;
+  NSAttributedString *_seletedAttributedTitle;
```
Comment (rahul-malik) [inline AsyncDisplayKit/ASButtonNode.mm]: spelling typo here. should be _selectedAttributedTitle

Comment (binl) [inline AsyncDisplayKit/ASButtonNode.mm]: =。= dang...

**Pull request merged**